tests/cpp/test_multidevice_sharding.cpp

```diff
@@ -745,4 +745,307 @@ TEST_F(MultiDeviceTest, ReorderDIDToFront) {
       __FILE__);
 }
 
+TEST_F(MultiDeviceTest, TransformPropagatorSplitReshape) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  const int d = communicator_->size();
+  const int64_t b = 2, s = 2, h = 4, e = 3;
+
+  TensorView* in = makeContigConcreteTensor(
+      {b, s, d * h * e}); // in: loop domain: {b, s, d*h*e}
+  TensorView* out = reshape(
+      in,
+      {b, s, d * h * e},
+      {b, s, d * h, e}); // out: loop domain: {b, s, d*h, e}
+
+  fusion->addInput(in);
+  fusion->addOutput(out);
+
+  auto mesh = DeviceMesh::createForNumDevices(d);
+
+  // Propagate transform from reshaped output to input.
+  // Without this propagation, the two DID axes on `in` and `out` will not be
+  // mapped in together in ID model. This causes scheduling to fail due to
+  // resharding.
+  TransformPropagator propagator_c2p(out);
+  MaxLogicalDomainInfoSpanningTree(out).traverse(&propagator_c2p);
+  // in: loop domain: {b, s, d*h, e} after transform propagation
```

Comment:
Does propagateShardings in this file fail on split reshapes?

Reply:
I've kept the example where we propagate reshapes before loop split in the case we decide to go that route. Hence, keeping this example from PR #3875 .